lib/go/test/flow_dkg_test.go

```diff
@@ -566,8 +566,103 @@ func TestDKG(t *testing.T) {
 		)
 	})
 
+	t.Run("Should not be able to set the safe threshold while the DKG is enabled", func(t *testing.T) {
+
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateSetSafeThresholdScript(env), DKGAddress)
+
+		err := tx.AddArgument(CadenceUFix64("0.90"))
+		require.NoError(t, err)
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, DKGAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), DKGSigner},
+			true,
+		)
+
+	})
+
+	t.Run("Admin Should be able to force stop the DKG", func(t *testing.T) {
+
+		tx := createTxWithTemplateAndAuthorizer(b, templates.GenerateForceStopDKGScript(env), DKGAddress)
+
+		signAndSubmit(
+			t, b, tx,
+			[]flow.Address{b.ServiceKey().Address, DKGAddress},
+			[]crypto.Signer{b.ServiceKey().Signer(), DKGSigner},
+			false,
+		)
+	})
+
+	t.Run("Should not be able to set the safe threshold less than 0.5 or greater than 1", func(t *testing.T) {
```

Comment:
I guess the 0.5 should be adjusted to the native threshold.

Reply:
Updated